src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1515 @@
+        heads, index = find_pruneable_heads_and_indices(heads, self.num_heads, self.head_dim, self.pruned_heads)
+
+        # Prune linear layers
+        self.Wqkv = prune_qkv_linear_layer(self.Wqkv, index)
+        self.Wo = prune_linear_layer(self.Wo, index, dim=1)
+
+        # Update hyper params and store pruned heads
+        self.num_heads = self.num_heads - len(heads)
+        self.all_head_size = self.head_dim * self.num_heads
+        self.pruned_heads = self.pruned_heads.union(heads)
+
+    def _init_weights(self, reset_params: bool = False):
+        _init_modernbert_weights(self.config, self.Wqkv, module_type=ModernBertModuleType.in_module)
+        _init_modernbert_weights(self.config, self.Wo, module_type=ModernBertModuleType.out_module)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        position_ids: Optional[torch.LongTensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        cu_seqlens: Optional[torch.Tensor] = None,
+        max_seqlen: Optional[int] = None,
+        output_attentions: Optional[bool] = False,
+        **kwargs,
+    ) -> torch.Tensor:
+        """Perform self-attention.
+
+        There are two attention implementations supported: PyTorch's SDPA attention and Flash Attention 2.
+
+        The arguments are unpadded. The SDPA implementation of attention requires padded arguments while the
+        Flash Attention implementation does not. If using SDPA we first call `pad_input`. Once we compute
+        attention, we re-unpad our outputs for the other layers. The pad/unpad operations add overhead, but not
+        sending pad tokens through ffs saves compute.
+
+        Args:
+            hidden_states: (total_nnz, dim)
+            cu_seqlens: (batch + 1,)
+            max_seqlen: int
+            indices: (total_nnz,)
+            attn_mask: (batch, max_seqlen)
+
+        Returns:
+            attention: (total_nnz, dim)
+        """
+        qkv = self.Wqkv(hidden_states)
+
+        attn_kwargs = {
+            "output_attentions": output_attentions,
+            "dim": self.all_head_size,
+        }
+        if self.config._attn_implementation == "flash_attention_2":
+            bs = hidden_states.shape[0]
+            qkv = qkv.view(-1, 3, self.num_heads, self.head_dim)
+
+            attn_kwargs.update(
+                {
+                    "local_attention": self.local_attention,
+                    "cu_seqlens": cu_seqlens,
+                    "max_seqlen": max_seqlen,
+                    "bs": bs,
+                }
+            )
+        else:
+            bs, seqlen = hidden_states.shape[:2]
+            qkv = qkv.view(bs, seqlen, 3, self.num_heads, self.head_dim)
+
+            attn_kwargs.update(
+                {
+                    "position_ids": position_ids,
+                    "attention_mask": attention_mask,
+                    "bs": bs,
+                    "seqlen": seqlen,
+                }
+            )
+
```

Comment:
same here, the if else is useless as you can just pass kwargs and the function can ignore them if needed

Reply:
Solved in 986c6feb9990b4e49c3ed810005e0855dd1c8b4f